src/Shortcodes/Shortcode.php

```diff
@@ -0,0 +1,205 @@
+<?php
+namespace ADM\WPPlugin\Shortcodes;
+
+use ADM\WPPlugin as ADMWPP;
+use ADM\WPPlugin\Oauth2;
+use ADM\WPPlugin\Settings;
+
+use Administrate\PhpSdk\GraphQl\Client as SDKClient;
+
+if (! class_exists('Shortcode')) {
+  /**
+   * Description of shortcodes
+   *
+   * @author
+   */
+    class Shortcode
+    {
+        protected static $instance;
+
+        function __construct()
+        {
+            if (file_exists('../../../../../wp-load.php')) {
+                require_once('../../../../../wp-load.php');
+            }
+
+            // Add all actions and filters
+            $this->addFilters();
+            $this->addActions();
+            $this->addShortcodes();
+        }
+
+        /**
+         * Static Singleton Factory Method
+         * Return an instance of the current class if it exists
+         * Construct a new one otherwise
+         *
+         * @return Course object
+         * */
+        public static function instance()
+        {
+            if (!isset(self::$instance)) {
+                $className = __CLASS__;
+                self::$instance = new $className;
+            }
+
+            return self::$instance;
+        }
```

Comment:
do we really need it ? can it be replaced with a constructor ?

Reply:
@alihabib-administrate creating the instance of this class is needed to fire up the creation of the filters / action / shortcodes. 
We do have a __construct method for this class also that is called upon instance creation by `new $className`
In Main.php we are instantiating the shortcode class.